test/jasmine/tests/polar_test.js

```diff
@@ -628,6 +631,35 @@ describe('Test relayout on polar subplots:', function() {
         .catch(failTest)
         .then(done);
     });
+
+    it('should not attempt to draw radial axis when *polar.hole* is set to 1', function(done) {
+        var gd = createGraphDiv();
+
+        var queries = [
+            '.radial-axis', '.radial-grid', '.radial-line > line',
+            '.radialdrag', '.radialdrag-inner'
+        ];
+
+        function _assert(msg, showing) {
+            var exp = showing ? null : 'none';
+            var sp3 = d3.select(gd).select('.polarlayer > .polar');
+            queries.forEach(function(q) {
+                assertNodeDisplay(sp3.selectAll(q), [exp], msg + ' ' + q);
+            });
+        }
+
+        Plotly.plot(gd, [{
+            type: 'scatterpolar',
+            r: ['a', 'b', 'c']
+        }])
+        .then(function() { _assert('base', true); })
+        .then(function() { return Plotly.relayout(gd, 'polar.hole', 1); })
+        .then(function() { _assert('hole=1', false); })
```

Comment:
So it doesn't fail... does it work? ie draws all data on the rim regardless of radial value (within the radial range)? Do you want to include this in a mock?

Reply:
Here's `hole=1` subplot:

![peek 2018-09-17 17-11](https://user-images.githubusercontent.com/6675409/45650579-bb02fd80-ba9c-11e8-9cce-b780c4550461.gif)

where angular drag still drag still works.

> ie draws all data on the rim regardless of radial value (within the radial range)?

it shows no data points.

> Do you want to include this in a mock?

done in https://github.com/plotly/plotly.js/pull/3021/commits/cb5b3368ed47f1db3904543006e07bba37d8f956

Reply:
> it shows no data points.

This comment is nonblocking (perhaps add to #2255 if you don't want to address it here):

With `cliponaxis: false` (which is the default anyway for polar) I would have expected it to show the data - and it currently doesn't, even if you explicitly specify a range. The argument for this behavior is to have `hole: 1` match `hole: 0.9999` as closely as possible. Perhaps not useful, though I guess I could see someone wanting to just show where their data is on an angular scale without involving a radial scale at all. That said they can always just use `hole: 0.9999` as a workaround, so it's not *really* any missing functionality... but it feels hacky.

Reply:
> The argument for this behavior is to have `hole: 1` match `hole: 0.9999` as closely as possible.

Ah I see. Good point.

> Perhaps not useful, though I guess I could see someone wanting to just show where their data is on an angular scale without involving a radial scale at all

I'll wait until someone complains about it (maybe @nicolaskruchten ?) before addressing this. Now in -> https://github.com/plotly/plotly.js/issues/2255#issuecomment-422409171